### ISSUE TITLE
TypeError: remove_pricing_rule_for_item() missing 1 required positional argument: 'pricing_rules'

### ISSUE THREAD

**Issue by Govind-Jangid** [frappe/erpnext#39912]
### Information about bug

If we change the qty in Sales Invoice while creating Sales Invoice from Sales Order, then this error occurred.

### Module

accounts

### Version

<img width="898" alt="image" src="https://github.com/frappe/erpnext/assets/63184660/b1ef158a-053c-4d1c-8db9-265bcd9862b5">

### Installation method

manual install

### Relevant log output / Stack trace / Full Error Message.

```shell
Traceback (most recent call last):
  File "apps/frappe/frappe/app.py", line 110, in application
    response = frappe.api.handle(request)
  File "apps/frappe/frappe/api/__init__.py", line 49, in handle
    data = endpoint(**arguments)
  File "apps/frappe/frappe/api/v1.py", line 36, in handle_rpc_call
    return frappe.handler.handle()
  File "apps/frappe/frappe/handler.py", line 49, in handle
    data = execute_cmd(cmd)
  File "apps/frappe/frappe/handler.py", line 85, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "apps/frappe/frappe/__init__.py", line 1682, in call
    return fn(*args, **newargs)
  File "apps/frappe/frappe/utils/typing_validations.py", line 31, in wrapper
    return func(*args, **kwargs)
TypeError: remove_pricing_rule_for_item() missing 1 required positional argument: 'pricing_rules'
```

**Comment by SamuelHelbling** [comment]
I have the same problem in Supplier Quotation

```
  Traceback (most recent call last):
  File "apps/frappe/frappe/app.py", line 110, in application
    response = frappe.api.handle(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/api/__init__.py", line 49, in handle
    data = endpoint(**arguments)
           ^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/api/v1.py", line 36, in handle_rpc_call
    return frappe.handler.handle()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/handler.py", line 49, in handle
    data = execute_cmd(cmd)
           ^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/handler.py", line 85, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/__init__.py", line 1682, in call
    return fn(*args, **newargs)
           ^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/utils/typing_validations.py", line 31, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
TypeError: remove_pricing_rule_for_item() missing 1 required positional argument: 'pricing_rules'
```

**Comment by perotom** [comment]
I have the same error with Delivery Note:
```
Traceback (most recent call last):
  File "apps/frappe/frappe/app.py", line 110, in application
    response = frappe.api.handle(request)
  File "apps/frappe/frappe/api/__init__.py", line 49, in handle
    data = endpoint(**arguments)
  File "apps/frappe/frappe/api/v1.py", line 36, in handle_rpc_call
    return frappe.handler.handle()
  File "apps/frappe/frappe/handler.py", line 49, in handle
    data = execute_cmd(cmd)
  File "apps/frappe/frappe/handler.py", line 85, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "apps/frappe/frappe/__init__.py", line 1682, in call
    return fn(*args, **newargs)
  File "apps/frappe/frappe/utils/typing_validations.py", line 31, in wrapper
    return func(*args, **kwargs)
TypeError: remove_pricing_rule_for_item() missing 1 required positional argument: 'pricing_rules'
 
```

**Comment by SvbZ3r0** [comment]
Closed by #39999

**Comment by casesolved-co-uk** [comment]
I have the same error with Purchase Receipt:

> Traceback (most recent call last):
>   File "apps/frappe/frappe/app.py", line 110, in application
>     response = frappe.api.handle(request)
>   File "apps/frappe/frappe/api/__init__.py", line 49, in handle
>     data = endpoint(**arguments)
>   File "apps/frappe/frappe/api/v1.py", line 36, in handle_rpc_call
>     return frappe.handler.handle()
>   File "apps/frappe/frappe/handler.py", line 49, in handle
>     data = execute_cmd(cmd)
>   File "apps/frappe/frappe/handler.py", line 85, in execute_cmd
>     return frappe.call(method, **frappe.form_dict)
>   File "apps/frappe/frappe/__init__.py", line 1682, in call
>     return fn(*args, **newargs)
>   File "apps/frappe/frappe/utils/typing_validations.py", line 31, in wrapper
>     return func(*args, **kwargs)
> TypeError: remove_pricing_rule_for_item() missing 1 required positional argument: 'pricing_rules'
